src/liballoc/btree/map.rs

```diff
@@ -1480,6 +1480,16 @@ impl<'a, K, V> Iterator for Range<'a, K, V> {
             unsafe { Some(self.next_unchecked()) }
         }
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        if self.front == self.back {
+            (0, Some(0))
+        } else {
+            // There doesn't seem to be any way to get the size of the whole tree
+            // from a NodeRef.
+            (1, None)
+        }
+    }
```

Comment:
What is the utility of a size hint like this? Would it be better to just not define it at all?

Reply:
I think it's useful to have it, because it may be implemented in the future. `size_hint` is easily forgotten unless it's in the code.

Reply:
I'm not sure I buy that form of reasoning.

Reply:
In this particular case I buy it because `(1, None)` in some cases is still more information than the default `size_hint` of `(0, None)`.

But https://github.com/rust-lang/rust/pull/49552 contains some `size_hint` implementations that literally just return `(0, None)`. For those cases I don't think I buy it.

Reply:
Well in particular, the bit about "because it may be implemented in the future." I'd rather not add speculative impls?

Reply:
@BurntSushi would just adding `// FIXME(#49205): Add size_hint` be better? I just want to make sure it's not forgotten.

Reply:
@Ixrec I don't know what you're talking about? None of the implementations I wrote return `(0, None)`, or even this controversial `(_, None)`... Please comment on that PR if you see any specific issues, maybe you see something I don't.

Reply:
The other option is to add a `length` field like _every other BTree iterator_. But that isn't a trivial thing, which is what this PR is for.

Reply:
I'm just going to move forward with a FIXME. There's no consensus about what to do.